crates/core/c8y_api/src/http_proxy.rs

```diff
@@ -17,24 +18,37 @@ use tracing::info;
 pub struct C8yEndPoint {
     pub c8y_host: String,
     pub device_id: String,
-    pub c8y_internal_id: String,
+    pub token: Option<String>,
+    pub devices_internal_id: HashMap<String, String>,
 }
 
 impl C8yEndPoint {
     pub fn new(c8y_host: &str, device_id: &str, c8y_internal_id: &str) -> C8yEndPoint {
+        let mut devices_internal_id = HashMap::new();
+        if !c8y_internal_id.is_empty() {
```

Comment:
I don't think there is a single place in the code where `C8yEndPoint::new()` is called with a meaningful internal id.

=> I would simply get rid of this 3rd argument.

Reply:
done